packages/sdk/src/cosmwasmclient.spec.ts

```diff
@@ -73,6 +73,40 @@ describe("CosmWasmClient", () => {
         sequence: 0,
       });
     });
+
+    it("returns zeros for missing accounts", async () => {
+      pendingWithoutCosmos();
+      const client = CosmWasmClient.makeReadOnly(httpUrl);
+      const missing = makeRandomAddress();
+      expect(await client.getNonce(missing)).toEqual({
+        accountNumber: 0,
```

Comment:
This is a strange behaviour. `accountNumber` is an on-chain value for the index of the account, right? So `accountNumber: 0` is of no use and even misleading. I think we should return `undefined` in this case

Reply:
If we want to avoid the undefined checks for the happy path, we should throw an error if the account does not exist.

Reply:
I made it undefined first, and it broke like 10 places, so I wondered about that. Throwing an error makes sense, as we only get the nonce when preparing to sign, and you cannot sign with a missing account

Reply:
I agree. Changed it to throw instead of returning invalid data.